asteroid/models/fasnet.py

```diff
@@ -0,0 +1,298 @@
+        )
+
+        # dual path processing is done we fold back
+        folded = F.fold(
+            unfolded,
+            (seq_length, 1),
+            kernel_size=(self.chunk_size, 1),
+            padding=(self.chunk_size, 0),
+            stride=(self.hop_size, 1),
+        )
+
+        # dividing to assure perfect reconstruction
+        folded = folded.squeeze(-1) / (self.chunk_size / self.hop_size)
+
+        # apply gating to output and scaling to -1 and 1)
+        folded = self.tanh(folded) * self.gate(folded)
+        folded = folded.view(batch_size, n_mics, self.n_src, -1, seq_length)
+
+        # beamforming
+        # convolving with all mic context --> Filter and Sum
+        all_mic_context = all_mic_context.unsqueeze(2).repeat(1, 1, 2, 1, 1)
+
+        all_bf_output = F.conv1d(
+            all_mic_context.view(1, -1, self.context * 2 + self.window),
+            folded.transpose(3, -1).contiguous().view(-1, 1, self.filter_dim),
+            groups=batch_size * n_mics * self.n_src * seq_length,
+        )
+        all_bf_output = all_bf_output.view(batch_size, n_mics, self.n_src, seq_length, self.window)
+
+        # fold back to obtain signal
+        all_bf_output = F.fold(
+            all_bf_output.reshape(
+                batch_size * n_mics * self.n_src, seq_length, self.window
+            ).transpose(1, -1),
+            (n_samples, 1),
+            kernel_size=(self.window, 1),
+            padding=(self.window, 0),
+            stride=(self.window // 2, 1),
+        )
+
+        bf_signal = all_bf_output.reshape(batch_size, n_mics, self.n_src, n_samples)
+
+        # we sum over mics after filtering (filters will realign the phase and time shift)
+        if valid_mics.max() == 0:
+            bf_signal = bf_signal.mean(1)
+        else:
+            bf_signal = [
+                bf_signal[b, : valid_mics[b]].mean(0).unsqueeze(0) for b in range(batch_size)
+            ]
+            bf_signal = torch.cat(bf_signal, 0)
+
+        return bf_signal
+
+    def get_config(self):
+        config = {
+            "enc_dim": self.enc_dim,
+            "feature_dim": self.feature_dim,
+            "hidden_dim": self.hidden_dim,
+            "n_layers": self.n_layers,
+            "n_src": self.n_src,
+            "window_ms": self.window_ms,
+            "stride": self.stride,
+            "context_ms": self.context_ms,
+            "samplerate": self.samplerate,
+            "tac_hidden_dim": self.tac_hidden_dim,
+            "norm_type": self.norm_type,
+            "chunk_size": self.chunk_size,
+            "hop_size": self.hop_size,
+            "bidirectional": self.bidirectional,
+            "rnn_type": self.rnn_type,
+            "dropout": self.dropout,
+            "use_tac": self.use_tac,
+        }
+
+        return config
+
+    @property
+    def sample_rate(self):
+        """Operating sample rate of the model (float)."""
+        raise self.samplerate
```

Comment:
```suggestion
        return self.samplerate
```

Reply:
Thanks. We'll have to pass `sample_rate` to `BaseModel` and remove that part anyway, but in the mean time, at least it makes sense, thanks.